go/vt/vtctl/grpcvtctldserver/server_test.go

```diff
@@ -436,6 +437,46 @@ func TestApplyVSchema(t *testing.T) {
 				},
 			},
 			shouldErr: false,
+		}, {
+			name: "strict unknown params",
+			req: &vtctldatapb.ApplyVSchemaRequest{
+				Keyspace: "testkeyspacesharded",
+				VSchema: &vschemapb.Keyspace{
+					Sharded: true,
+					Vindexes: map[string]*vschemapb.Vindex{
+						"lookup1": {
+							Type: "lookup",
+							Params: map[string]string{
+								"hello":   "world",
+								"goodbye": "world",
+							},
+						},
+					},
+				},
+				SkipRebuild: true,
+				Strict:      true,
+			},
+			exp: &vtctldatapb.ApplyVSchemaResponse{
+				VSchema: &vschemapb.Keyspace{
+					Sharded: true,
+					Vindexes: map[string]*vschemapb.Vindex{
+						"lookup1": {
+							Type: "lookup",
+							Params: map[string]string{
+								"hello":   "world",
+								"goodbye": "world",
+							},
+						},
+					},
+				},
+				UnknownVindexParams: map[string]*vtctldatapb.ApplyVSchemaResponse_ParamList{
+					"lookup1": {
+						Params: []string{"goodbye", "hello"},
+					},
+				},
+			},
+			shouldErr: true,
+			err:       "unknown vindex params: lookup1 (goodbye, hello)",
```

Comment:
Are we sorting the keys and values so that the tests are not flaky?

Reply:
For example, I really should have done that here in a recent PR:
```
diff --git a/go/vt/vttablet/tabletmanager/vreplication/controller_plan.go b/go/vt/vttablet/tabletmanager/vreplication/controller_plan.go
index 0273f05ab4..f8cb12ec10 100644
--- a/go/vt/vttablet/tabletmanager/vreplication/controller_plan.go
+++ b/go/vt/vttablet/tabletmanager/vreplication/controller_plan.go
@@ -18,6 +18,7 @@ package vreplication

 import (
        "fmt"
+       "slices"
        "strings"

        "vitess.io/vitess/go/constants/sidecar"
@@ -122,6 +123,7 @@ func columnsAsCSV(columns []*sqlparser.ColName) string {
        for i := range columns {
                colsForError[i] = columns[i].Name.String()
        }
+       slices.Sort(colsForError)
        return strings.Join(colsForError, ", ")
 }
```

Reply:
We are already sorting by [keys](https://github.com/vitessio/vitess/pull/14862/files#diff-ad37010accc5a29c6751d18124328bb882457e8b2527db688b854c81fe23861cR3407) and [values](https://github.com/vitessio/vitess/blob/4bb84ed3ccd2ade17ba3cc876c76762f35f4c5b1/go/vt/vtgate/vindexes/vindex.go#L252).